tools/src/whitenoise.sh

```diff
@@ -1,7 +1,8 @@
 #!/bin/bash
 TRAILING_WHITESPACE=0
 # only check changed docs in pr
-for file in $(git diff --cached --name-only --diff-filter=ACRMTU $GITHUB_BASE_REF | grep "\.md"); do
+git fetch
+for file in $(git diff --cached --name-only --diff-filter=ACRMTU origin/$GITHUB_BASE_REF| grep "\.md"); do
```

Comment:
thinking out loud, why isn't `git diff --name-only` good enough here ?

Reply:
The issue here is that the GitHub Actions environment does not have the branch information for Git, which causes an error when using `git diff` to compare the differences. To resolve this, I simply added a `git fetch` command to retrieve the branch information from the remote repository. For further debugging information, you can refer to this link: [https://github.com/zaunist/gateway/actions/runs/6262614026/job/17005194717](https://github.com/zaunist/gateway/actions/runs/6262614026/job/17005194717).

Reply:
@arkodg If we don't provide branch information, how can we compare the differences between the fork and main branches? I'm not quite understanding this point. Could you provide an example, please?